### PR TITLE
feat: add support for picking of point clouds

### PR DESCRIPTION
**Pull request by larsmoa**

### CODE REVIEW

File: viewer/src/public/migration/Cognite3DViewer.ts [hunk 1]
```diff
@@ -893,33 +896,60 @@ export class Cognite3DViewer {
...
-      coords,
+    const input: IntersectInput = {
+      normalizedCoords: coords,
       camera: this.camera,
       renderer: this.renderer
-    });
+    };
+    const cadResults = intersectCadNodes(cadNodes, input);
+    const pointCloudResults = intersectPointClouds(pointCloudNodes, input);
+    console.log(pointCloudResults);
```
Comment (maksnester) [inline viewer/src/public/migration/Cognite3DViewer.ts:914]:
```suggestion
```
I'm quite confused why does linter allow it...

File: viewer/src/public/migration/Cognite3DViewer.ts [hunk 2]
```diff
@@ -893,33 +896,60 @@ export class Cognite3DViewer {
...
           };
-          return intersection;
+          intersections.push(intersection);
         }
       }
     }
 
-    return null;
+    intersections.sort((a, b) => a.distanceToCamera - b.distanceToCamera);
+    console.log(intersections);
```
Comment (maksnester) [inline viewer/src/public/migration/Cognite3DViewer.ts:951]:
```suggestion
```

File: documentation/docs/examples/pointcloud-intersections.mdx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+The following example detects intersections at clicked positions and marks the intersected positions
+using a red sphere.
+
+```js runnable
+viewer.on('click', event => {
+  const intersection = viewer.getIntersectionFromPixel(
+    event.offsetX, event.offsetY
+  );
+  if (intersection) {
+    console.log(intersection);
```
Comment (maksnester) [inline documentation/docs/examples/pointcloud-intersections.mdx:26]:
```suggestion
```

File: Cargo.lock [hunk 4]
```diff
@@ -46,6 +46,15 @@ dependencies = [
  "num-traits",
 ]
 
+[[package]]
```
Comment (maksnester) [inline Cargo.lock:49]: Why this file has changed?

**Review by maksnester — COMMENTED** [review]
It looks good to me in general, but I'll leave it for @christjt approval and review of 3d parts.

File: Cargo.lock (hunk 4 above)
Comment (larsmoa) [reply Cargo.lock]: Reverted all changes to this file.

**Review by christjt — APPROVED** [review]

**Pull request merged**